core/config/v2/types.go

```diff
@@ -71,14 +66,27 @@ type Secrets struct {
 
 func (s *Secrets) ValidateConfig() (err error) {
 	if s.DatabaseURL == nil || (*url.URL)(s.DatabaseURL).String() == "" {
-		err = multierr.Append(err, ErrEmpty{Name: "Database URL", Msg: "must be provided and non-empty"})
+		err = multierr.Append(err, ErrEmpty{Name: "DatabaseURL", Msg: "must be provided and non-empty"})
+	} else {
+		if verr := config.ValidateDBURL((url.URL)(*s.DatabaseURL)); verr != nil {
+			err = multierr.Append(err, ErrInvalid{Name: "DatabaseURL", Value: "*****", Msg: dbURLPasswordComplexity(verr)})
+		}
+	}
+	if s.DatabaseBackupURL != nil {
+		if verr := config.ValidateDBURL((url.URL)(*s.DatabaseBackupURL)); verr != nil {
+			err = multierr.Append(err, ErrInvalid{Name: "DatabaseBackupURL", Value: "*****", Msg: dbURLPasswordComplexity(verr)})
+		}
 	}
 	if s.KeystorePassword == nil || *s.KeystorePassword == "" {
-		err = multierr.Append(err, ErrEmpty{Name: "Keystore Password", Msg: "must be provided and non-empty"})
+		err = multierr.Append(err, ErrEmpty{Name: "KeystorePassword", Msg: "must be provided and non-empty"})
 	}
 	return err
 }
 
+func dbURLPasswordComplexity(err error) string {
```

Comment:
Will it fail to boot now if this is insufficiently complex? This could surprise nops (even though we have been logging warnings for a couple of releases now). Maybe we ought to start making a detailed list of breaking changes?

Reply:
Perhaps, but wouldn't it be more practical for node ops to dump their specific config and then validate?

Reply:
Yes, I think this workflow will be ok.